### PR TITLE
Prepare v0.45.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes https://github.com/open-telemetry/opentelemetry-collector/issues/4875
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Review by tigrannajaryan — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ vendor
 **/_build
 dist/
 .idea
+.generated-yaml/
```
Comment (bogdandrutu) [inline .gitignore:5]: why this change?

Comment (jpkrohling) [reply .gitignore:5]: When testing it locally, I noticed that this directory existed and was added to my commit. I think it's not needed anymore after we replaced shell scripts with Go, but there might be workspaces out there with it.
